FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,35 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				if response.statusCode == 200, let json = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(FeedImageMapper.map(json.items)))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+private struct Root: Decodable {
+	let items: [Item]
+}
+
+private struct Item: Decodable {
+	let image_id: UUID
+	let image_desc: String?
+	let image_loc: String?
+	let image_url: URL
+}
+
+private class FeedImageMapper {
```

Comment:
Since `FeedImageMapper` is not supposed to be subclassed, it should be marked as final.
```suggestion
private final class FeedImageMapper {
```